src/persist-client/src/internal/service.proto

```diff
@@ -0,0 +1,40 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+syntax = "proto3";
+
+package mz_persist_client.internal.service;
+
+message ProtoPushDiff {
+    string shard_id = 1;
+    uint64 seqno = 2;
+    bytes diff = 3;
+}
+
+message ProtoSubscribe {
+    string shard = 1;
+}
+
+message ProtoUnsubscribe {
+    string shard = 1;
+}
+
+message ProtoPubSubMessage {
+    bytes timestamp = 1;
```

Comment:
I think the timestamp is probably better serialized as a u64 seconds and a u64 nanos